### ISSUE TITLE
Unable to setup SASL enabled memcached instance on mac

### ISSUE THREAD

**Issue by konnectnishant** [memcached/memcached#263]
Hi team,
Does anyone of issue, connecting SASL enabled memcache.

Here are the few steps I followed to install memcached with SASL on my mac but when I connect using java client (attached) it’s fails with password verification failed error? Any one can help! Thanks!

Nishants-MacBook-Pro-2:~ nishant_jain$ brew install memcached --enable-sasl-pwdb
Nishants-MacBook-Pro-2:~ nishant_jain$ echo "mech_list: plain" > memcached.conf
Nishants-MacBook-Pro-2:~ nishant_jain$ echo "nishant:pass" > /tmp/memcached-sasl-db
Nishants-MacBook-Pro-2:~ nishant_jain$ export MEMCACHED_SASL_PWDB=/tmp/memcached-sasl-db
Nishants-MacBook-Pro-2:~ nishant_jain$ export SASL_CONF_PATH=`pwd`/memcached.conf

Nishants-MacBook-Pro-2:~ nishant_jain$ /usr/local/opt/memcached/bin/memcached -S -vv -p 8010
Initialized SASL.
slab class   1: chunk size        96 perslab   10922
slab class   2: chunk size       120 perslab    8738

Memcache terminal dump:

<27 new binary client connection.
<27 Read binary protocol data:
<27    0x80 0x21 0x00 0x05
<27    0x00 0x00 0x00 0x00
<27    0x00 0x00 0x00 0x12
<27    0x00 0x00 0x00 0x11
<27    0x00 0x00 0x00 0x00
<27    0x00 0x00 0x00 0x00
authenticated() in cmd 0x21 is true
mech:  ``PLAIN'' with 13 bytes of data
INFO: User <nishant@Nishants-MacBook-Pro-2.local> failed to authenticate
SASL (severity 2): Password verification failed
sasl result code:  -20
Unknown sasl response:  -20
>27 Writing an error: Auth failure.

Intellij Exception:

2017-02-23 16:42:36.510 INFO net.spy.memcached.MemcachedConnection:  Added {QA sa=localhost/127.0.0.1:11211, #Rops=0, #Wops=0, #iq=0, topRop=null, topWop=null, toWrite=0, interested=0} to connect queue
2017-02-23 16:42:36.607 ERROR net.spy.memcached.protocol.binary.SASLAuthOperationImpl:  Error:  Auth failure.
2017-02-23 16:42:36.608 WARN net.spy.memcached.protocol.binary.BinaryMemcachedNodeImpl:  Discarding partially completed op: SASL auth operation
2017-02-23 16:42:36.609 INFO net.spy.memcached.MemcachedConnection:  Reconnection due to exception handling a memcached operation on {QA sa=localhost/127.0.0.1:11211, #Rops=0, #Wops=0, #iq=0, topRop=null, topWop=null, toWrite=0, interested=1}. This may be due to an authentication failure.
OperationException: SERVER: Auth failure.
	at net.spy.memcached.protocol.BaseOperationImpl.handleError(BaseOperationImpl.java:192)
	at net.spy.memcached.protocol.binary.OperationImpl.finishedPayload(OperationImpl.java:204)
	at net.spy.memcached.protocol.binary.SASLBaseOperationImpl.finishedPayload(SASLBaseOperationImpl.java:98)
	at net.spy.memcached.protocol.binary.OperationImpl.readPayloadFromBuffer(OperationImpl.java:196)
	at net.spy.memcached.protocol.binary.OperationImpl.readFromBuffer(OperationImpl.java:139)
	at net.spy.memcached.MemcachedConnection.readBufferAndLogMetrics(MemcachedConnection.java:861)
	at net.spy.memcached.MemcachedConnection.handleReads(MemcachedConnection.java:840)
	at net.spy.memcached.MemcachedConnection.handleReadsAndWrites(MemcachedConnection.java:720)
	at net.spy.memcached.MemcachedConnection.handleIO(MemcachedConnection.java:683)
	at net.spy.memcached.MemcachedConnection.handleIO(MemcachedConnection.java:436)
	at net.spy.memcached.MemcachedConnection.run(MemcachedConnection.java:1446

[memcache-java.txt](https://github.com/memcached/memcached/files/800309/memcache-java.txt)

**Comment by konnectnishant** [comment]
Are there any step by step documentation on how to enable SASL on mac. I had hard time figuring it out on web, as there are pretty less documentation on it. Not sure if there is bug in client or their is some set up problem I'm having.

**Comment by konnectnishant** [comment]
Yay! I got the issue. Couch base client automatically adds hostname at the end of the username. In my case when I set the username as myuser:mypassword but when I pass myuser in my java code, memcacheclient pass that information as myuser@Nishants-MacBook-Pro-2.local and that's where the mismatch was happening. To solve that I added the complete username with local host when creating username and password.

echo "nishant:pass" > /tmp/memcached-sasl-db

got replaced with

echo "nishant@Nishants-MacBook-Pro-2.local:pass" > /tmp/memcached-sasl-db
